.pre-commit-config.yaml

```diff
@@ -64,6 +41,29 @@ repos:
     hooks:
       - id: black
 
+  - repo: https://github.com/pre-commit/pygrep-hooks
```

Comment:
I don't know there's any reason to move `pygrep` as it's just a static set of regexes. Feel free to convince me otherwise (if there's a strong reason behind this). I'm fine with moving yesqa lower, however.

Reply:
The flipside of this is I think running `yesqa` after `flake8` might cause less surprising behaviors. I'd be interested in that change on that PR.

Reply:
Ah yes, I see that the diff seems to indicate that I've moved `pygrep` and `flake8` down, but really what I've actually done is move `isort` and `black` up, since they are not static.

The ordering was:

- pre-commit-hooks (changes code)
- pygrep (static)
- yesqa (static)
- pyupgrade (changes code)
- isort (changes code)
- black (changes code)
- flake8 (static)

and I changed it to

- pre-commit-hooks (changes code)
- pyupgrade (changes code)
- isort (changes code)
- black  (changes code)
- pygrep (static)
- yesqa (static)
- flake8 (static)

The order of the code-changing hooks relative to themselves is the same, and the same goes for static checks.

Does that make sense? 🙂 The github diff viewer doesn't really do this justice so my bad for not highlighting it earlier

Reply:
Do you want me to change it to this? (flake8 running before yesqa)

pre-commit-hooks (changes code)
pyupgrade (changes code)
isort (changes code)
black (changes code)
pygrep (static)
flake8 (static)
yesqa (static)

Reply:
yesqa is actually a code changer. It removes any noqa that does not change the output of flake8. So by your logic here it should probably go before isort but after pyupgrade?

Reply:
Oh, I see! Out of all of the hooks, yesqa is the one I don't use myself. So maybe:

```
1. pre-commit-hooks
2. pyupgrade
3. yesqa 
4. isort 
5. black 
```

Would that make sense, or would another order make more sense to you?

Do you still agree with the premise that putting code altering hooks before static hooks (flake8 in particular) is a good thing, to eliminate potential false positives?

Reply:
I have to admit I'm not entirely convinced that this matters in practice, but I'm happy to defer to your greater experience. I think the order described above should be just fine and I'd be happy to accept the change.

Reply:
That's fair enough. I'll close this then for now 🙂